### DISCUSSION TITLE
Failing Healthchecks?

### DISCUSSION THREAD

**Discussion by jdsmi** [mbround18/valheim-docker#263] [Help!]
I have 2 servers running and one is failing health checks after being up for about 15 minutes or so. Not sure what might be the cause and nothing is indicating failing (logs don't show anything)

**Comment by CosmicHorrorDev** [comment]
What are the system specifications (primarily the amount of RAM), and are they using entirely separate mounts as well?

**Reply by jdsmi** [reply]
Yeah sorry I made this TOO vague in hopes that it was a known issue,

It's running in a GCP instance, 2vCPU, 4GB RAM

saves and backups are mounted to a persistent disk (50GB SSD)

**Reply by jdsmi** [reply]
another thing to note is it utilizes an imported save from a windows system, not that it should matter, the save is ~80MB

**Reply by CosmicHorrorDev** [reply]
4GB will definitely be pushing it for 2 servers. Can you switch it to 8GB and see if that _automagically_ fixes things?

I've ran 2 instances on my laptop without issue. Also sometimes the logs will indicate memory problems, but other times nothing shows up, so it can be a bit confusing to debug.

**Reply by jdsmi** [reply]
Sorry that's for each server not shared. I wasn't specific.

I tried recreating the instance to see if there was corruption or another resource error. I'll report back if there are any issues with the new instance.

**Reply by CosmicHorrorDev** [reply]
Ah, yeah that should be sufficient. And the two servers are using separate mounts for everything too?

**Reply by jdsmi** [reply]
Yeah everything is completely isolated, they each get dedicated disks for storage.

The new instance has been running for nearly 50 minutes without issue. It was probably an issue with the old instance.

**Reply by jdsmi** [reply]
well it seems saying it's fine made it not fine. It's back to failing health checks again.

For what it's worth the backups are still saving and there's nothing in the logs indicating any issues. The server stops showing up on the server list when this happens as well.

**Reply by CosmicHorrorDev** [reply]
I'm not sure then. Hopefully someone else can chime in with some advice

**Reply by mbround18** [reply]
When you docker exec into the instance is the server process running?

**Reply by jdsmi** [reply]
It didn't look like it. It seemed to silently fail.

Since restarting the instance again it seems to be working. It's been up for a few hours with no issue.

I have `ps` in the container now so if it does go down I can check if the server is running or not.

**Reply by mbround18** [reply]
Can you run with DEBUG_MODE=1 and drop your logs from `/home/steam/valheim/logs` into a zip file here? or drop all of them here? (either works)

**Reply by jdsmi** [reply]
It apparently wasn't an issue with the container or odin, I found a buried line in the server logs that the process was killed because it was out of memory, the imported save file has many objects in it so apparently 4 GB isn't enough memory. I'll have to scale up the VM.

**Reply by slayer1551** [reply]
Did more ram fix it ? I have 4gb aws server and just started having it happen. It’s fine when people are on the server but after an hour of it being idle it becomes unhealthy and I have to restart the instance

**Reply by jdsmi** [reply]
> 
> 
> Did more ram fix it ? I have 4gb aws server and just started having it happen. It’s fine when people are on the server but after an hour of it being idle it becomes unhealthy and I have to restart the instance

Yeah it seemed to. The problem is the application just stops silently when it goes out of memory. You could have the container restart when it's unhealthy but that doesn't stop it from crashing unless you increase the memory.
